server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -174,21 +181,37 @@ pub struct ListAttemptsByEndpointQueryParameters {
 
 // Applies filters common to [`list_attempts_by_endpoint`] and [`list_attempts_by_msg`]
 fn list_attempts_by_endpoint_or_message_filters(
-    query: Select<messageattempt::Entity>,
+    mut query: Select<messageattempt::Entity>,
     limit: u64,
-    iterator: Option<MessageAttemptId>,
+    iterator: Option<PaginationIterator<MessageAttemptId>>,
     status: Option<MessageStatus>,
     status_code_class: Option<StatusCodeClass>,
     event_types: Option<EventTypeNameSet>,
     channel: Option<EventChannel>,
 ) -> Select<messageattempt::Entity> {
-    let mut query = query
-        .order_by_desc(messageattempt::Column::Id)
-        .limit(limit + 1);
-
-    if let Some(iterator) = iterator {
-        query = query.filter(messageattempt::Column::Id.lt(iterator));
-    }
+    query = match iterator {
+        Some(PaginationIterator::Prev(id)) => query
+            .filter(
+                Condition::any().add(
+                    messageattempt::Column::Id.in_subquery(
+                        Query::select()
+                            .column(messageattempt::Column::Id)
+                            .and_where(messageattempt::Column::Id.gt(id))
+                            .order_by_columns(vec![(messageattempt::Column::Id, Order::Asc)])
+                            .limit(limit + 1)
+                            .to_owned(),
+                    ),
+                ),
+            )
+            .order_by_desc(messageattempt::Column::Id),
```

Comment:
You can move the `order_by_desc` outside of the match, it's used in all three options. Heck, you can just do it before the match even like where it was before and even keep the limit (because it doesn't matter).